From f719cc284ca8e70b34ad8ca9e1a5293cea5241ba Mon Sep 17 00:00:00 2001
From: Manoj Govindassamy <manojg@cloudera.com>
Date: Thu, 12 Jan 2017 11:01:30 -0800
Subject: [PATCH 2158/2848] CLOUDERA-BUILD. CDH-37372 Change DataNode logging
 level for slow BlockReceiver actions

BlockReceiver logs a WARN message for every IO operations relating to flush,
sync, write to mirror, write to disk, ack upstream that are taking more than
300ms. These slowness are generally seen in a cluster under heavy load. In
the past these WARN messages had led to several support calls and only
asked to be ignored in an otherwise healthy cluster.

Change-Id: I86b4bdee4e96c91c926d01f9cc9392f647a858be
---
 .../hadoop/hdfs/server/datanode/BlockReceiver.java |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockReceiver.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockReceiver.java
index 4d535b2..c29866d 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockReceiver.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockReceiver.java
@@ -416,9 +416,9 @@ void flushOrSync(boolean isSync) throws IOException {
     }
     long duration = Time.monotonicNow() - begin;
     if (duration > datanodeSlowLogThresholdMs) {
-      LOG.warn("Slow flushOrSync took " + duration + "ms (threshold="
-          + datanodeSlowLogThresholdMs + "ms), isSync:" + isSync + ", flushTotalNanos="
-          + flushTotalNanos + "ns");
+      LOG.info("Slow flushOrSync took " + duration + "ms (threshold="
+          + datanodeSlowLogThresholdMs + "ms), isSync:" + isSync
+          + ", flushTotalNanos=" + flushTotalNanos + "ns");
     }
   }
 
@@ -560,7 +560,7 @@ private int receivePacket() throws IOException {
         setLastSentTime(now);
         long duration = now - begin;
         if (duration > datanodeSlowLogThresholdMs) {
-          LOG.warn("Slow BlockReceiver write packet to mirror took " + duration
+          LOG.info("Slow BlockReceiver write packet to mirror took " + duration
               + "ms (threshold=" + datanodeSlowLogThresholdMs + "ms)");
         }
       } catch (IOException e) {
@@ -692,7 +692,7 @@ private int receivePacket() throws IOException {
           out.write(dataBuf.array(), startByteToDisk, numBytesToDisk);
           long duration = Time.monotonicNow() - begin;
           if (duration > datanodeSlowLogThresholdMs) {
-            LOG.warn("Slow BlockReceiver write data to disk cost:" + duration
+            LOG.info("Slow BlockReceiver write data to disk cost:" + duration
                 + "ms (threshold=" + datanodeSlowLogThresholdMs + "ms)");
           }
 
@@ -855,7 +855,7 @@ private void manageWriterOsCache(long offsetInBlock) {
         lastCacheManagementOffset = offsetInBlock;
         long duration = Time.monotonicNow() - begin;
         if (duration > datanodeSlowLogThresholdMs) {
-          LOG.warn("Slow manageWriterOsCache took " + duration
+          LOG.info("Slow manageWriterOsCache took " + duration
               + "ms (threshold=" + datanodeSlowLogThresholdMs + "ms)");
         }
       }
@@ -1533,7 +1533,7 @@ private void sendAckUpstreamUnprotected(PipelineAck ack, long seqno,
       upstreamOut.flush();
       long duration = Time.monotonicNow() - begin;
       if (duration > datanodeSlowLogThresholdMs) {
-        LOG.warn("Slow PacketResponder send ack to upstream took " + duration
+        LOG.info("Slow PacketResponder send ack to upstream took " + duration
             + "ms (threshold=" + datanodeSlowLogThresholdMs + "ms), " + myString
             + ", replyAck=" + replyAck);
       } else if (LOG.isDebugEnabled()) {
-- 
1.7.9.5

